herbert_core/utilities/misc/horace_memory.m

```diff
@@ -0,0 +1,77 @@
+function [phys_memory,free_memory]=horace_memory()
+%HORACE_MEMORY Simple system-indepentent utility used by Horace to identify
+% its type for proper auto-configuration
+%
+% Returns:
+% phys_memory  -- physical memory installed on the computer
+% free_memory  -- free memory available for utilites.
+%                 Returns these parameters empty if the memory
+%                 identification does not work.
+
+if ispc
+    [~,sys] = memory();
+    phys_memory = sys.PhysicalMemory.Total;
+    free_memory = sys.PhysicalMemory.Available;
+elseif isunix
+    if ismac
+        [nok,mem_string] = system('top -l 1 -n 0');
+        if ~nok
+            [phys_memory,free_memory] = parse_mac_mem_string(mem_string);
+        end
+    else % normal Unix
+        [nok,mem_string] = system('free | grep Mem');
+        if ~nok
+            [phys_memory,free_memory] = parse_mem_string(mem_string);
+        end
+    end
+    if nok
+        phys_memory = [];
+        free_memory = [];
+    end
+end
+
+function [phys_memory,free_memory] = parse_mem_string(mem_string)
+% Analyze result of free function on linux
+try
+    cont = regexp(mem_string,'\s+','split');
+    phys_memory =  1024*sscanf(cont{2},'%d');
+    free_memory =  1024*sscanf(cont{4},'%d');
+catch
+    phys_memory = [];
+    free_memory = [];
+
+end
```

Comment:
Would prefer `try-catch` outside of functions, and probably throw if `nok` (or `err ~= 0`  w/ refactor), defaulting to empty arrays

Reply:
ibid. Leave as it is.